### PR TITLE
DDF-3441 support ui color configuration

### PR DESCRIPTION
**Pull request by glenhein**
#### What does this PR do?
Allow the colors used by Intrique to be configured site-wide by the administrator and on a per-user basis in the Theme Settings.

#### Who is reviewing it? 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
@bdeining 
@troymohl 
#### Select relevant component teams: 
https://github.com/orgs/codice/teams/ui
@andrewkfiedler 
#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@lessarderic
@millerw8
#### How should this be tested? (List steps with links to updated documentation)
1) build and install
2) change a color value in the admin config and observe that the ui displays the new color, you may need to refresh the ui page and possibly delete the preferences using the Storage tab in your browsers developer tools
3) change a color value in the Theme settings and observe that the ui displays the new color, you may need to refresh the ui page and possibly delete the preferences using the Storage tab in your browsers developer tools

#### Any background context you want to provide?
The purpose of this PR is allow downstream projects to customize the colors used by the ui

#### What are the relevant tickets?
[DDF-3441](https://codice.atlassian.net/browse/DDF-3441)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests
#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by bdeining** [comment]
@glenhein any screenshots?

File: catalog/ui/catalog-ui-search/src/main/webapp/js/model/User.js [hunk 1]
```diff
@@ -159,7 +159,8 @@ define([
                 goldenLayoutMetacard: undefined,
                 goldenLayoutAlert: undefined,
                 theme: new Theme(),
-                animation: true
+                animation: true,
+                // TODO backgroundColor: _get(properties, 'backgroundColor', 'blue')
```
Comment (glenhein) [inline catalog/ui/catalog-ui-search/src/main/webapp/js/model/User.js]: i'll remove this

**Comment by glenhein** [comment]
@bdeining I could make a screenshot, but it would look like regular ddf :-)

**Comment by bdeining** [comment]
@glenhein i meant a screenshot demonstrating how changing this configuration changes the ui color scheme

**Comment by glenhein** [comment]
Admin config: 
screen shot 2017-11-10 at 9 29 00 am

**Comment by glenhein** [comment]
User config:
screen shot 2017-11-10 at 9 33 06 am

**Comment by bdeining** [comment]
screen shot 2017-11-10 at 9 59 22 am

**Comment by andrewkfiedler** [comment]
Commit pulled into https://github.com/codice/ddf/pull/2679

**Pull request closed without merging**